network/p2p/p2ptest/client.go

```diff
@@ -5,86 +5,50 @@ package p2ptest
 
 import (
 	"context"
-	"testing"
 	"time"
 
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/stretchr/testify/require"
-
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/network/p2p"
 	"github.com/ava-labs/avalanchego/snow/engine/common"
-	"github.com/ava-labs/avalanchego/snow/engine/enginetest"
-	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/avalanchego/utils/set"
 )
 
-// NewClient generates a client-server pair and returns the client used to
-// communicate with a server with the specified handler
-func NewClient(
-	t *testing.T,
-	ctx context.Context,
-	handler p2p.Handler,
-	clientNodeID ids.NodeID,
-	serverNodeID ids.NodeID,
-) *p2p.Client {
-	clientSender := &enginetest.Sender{}
-	serverSender := &enginetest.Sender{}
-
-	clientNetwork, err := p2p.NewNetwork(logging.NoLog{}, clientSender, prometheus.NewRegistry(), "")
-	require.NoError(t, err)
-
-	serverNetwork, err := p2p.NewNetwork(logging.NoLog{}, serverSender, prometheus.NewRegistry(), "")
-	require.NoError(t, err)
+var _ p2p.ClientInterface = (*Client)(nil)
 
-	clientSender.SendAppGossipF = func(ctx context.Context, _ common.SendConfig, gossipBytes []byte) error {
-		// Send the request asynchronously to avoid deadlock when the server
-		// sends the response back to the client
-		go func() {
-			require.NoError(t, serverNetwork.AppGossip(ctx, clientNodeID, gossipBytes))
-		}()
-
-		return nil
-	}
+// Client is a testing implementation of p2p.ClientInterface against a specified
+// server handler. The zero value of Client times out AppRequest and drops
+// outbound AppGossip messages.
+type Client struct {
+	NodeID       ids.NodeID
+	ServerNodeID ids.NodeID
```

Comment:
* `NodeID` -> `MyNodeID` seems strange because it's disambiguated by `Client.NodeID`
* `ServerNodeID` -> `PeerNodeID`/`PeerID` I am open to changing

Reply:
* `NodeID` -> `MyNodeID` seems unnecessary because `Client.NodeID` already disambiguates the name
* `ServerNodeID` -> `PeerNodeID` I could get behind... but I wouldn't be sure what to name `Client` as `Client` talking to `Server` seems more natural than `Client` talking to `Peer`. I guess we could call it `Sender`?

Reply:
Ya I think you're right `client.NodeID` should be clear. I prefer `PeerNodeID` since it's a p2p network rather than single server and updating `Handler` to `PeerHandler` or `ServerHandler` would make it a little more clear.